From: Armbian <build@armbian.com>
Date: Mon, 16 Dec 2024 12:00:00 +0900
Subject: [PATCH] arm: dts: sun50i-h616: Add H20 Pro 5G device tree

Add device tree file for Hechuang H20 Pro 5G board with H616 SoC.

---
 arch/arm/dts/sun50i-h616-h20-pro-5g.dts | 72 ++++++++++++++++++++++++++++++
 1 file changed, 72 insertions(+)
 create mode 100644 arch/arm/dts/sun50i-h616-h20-pro-5g.dts

diff --git a/arch/arm/dts/sun50i-h616-h20-pro-5g.dts b/arch/arm/dts/sun50i-h616-h20-pro-5g.dts
new file mode 100644
index 0000000000..0000000000
--- /dev/null
+++ b/arch/arm/dts/sun50i-h616-h20-pro-5g.dts
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (C) 2024 Armbian
+ */
+
+/dts-v1/;
+
+#include "sun50i-h616.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+
+/ {
+	model = "H20 Pro 5G";
+	compatible = "hechuang,h20-pro-5g", "allwinner,sun50i-h616";
+
+	aliases {
+		serial0 = &uart0;
+		ethernet0 = &emac0;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+	};
+
+	reg_vcc3v3: regulator-vcc3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "vcc3v3";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		regulator-always-on;
+	};
+};
+
+&i2c0 {
+	status = "disabled";
+};
+
+&mmc0 {
+	broken-cd;
+	bus-width = <4>;
+	vmmc-supply = <&reg_vcc3v3>;
+	vqmmc-supply = <&reg_vcc3v3>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc0_pins>;
+	status = "okay";
+};
+
+&mmc2 {
+	bus-width = <8>;
+	non-removable;
+	cap-mmc-hw-reset;
+	vmmc-supply = <&reg_vcc3v3>;
+	vqmmc-supply = <&reg_vcc3v3>;
+	pinctrl-names = "default";
+	pinctrl-0 = <&mmc2_pins>;
+	status = "okay";
+};
+
+&pio {
+	vcc-pc-supply = <&reg_vcc3v3>;
+	vcc-pf-supply = <&reg_vcc3v3>;
+	vcc-pg-supply = <&reg_vcc3v3>;
+};
+
+&uart0 {
+	status = "okay";
+};
+
+&emac0 {
+	phy-mode = "rmii";
+	status = "okay";
+};
+
+&hdmi {
+	status = "okay";
+};
+
+&de {
+	status = "okay";
+};
